### PR TITLE
Starting a group chat shouldn't show the section to fetch more messages.

### LINKED ISSUE #2379: starting a group chat shouldn't show the section to fetch more messages

### PR DESCRIPTION
**Pull request by pinebit**
Fixes #2379

### CODE REVIEW

**Review by richard-ramos — APPROVED** [review]
LGTM

File: ui/app/AppLayouts/Chat/ChatColumn/Message.qml [hunk 1]
```diff
@@ -283,14 +283,21 @@ Item {
         id: fetchMoreMessagesButtonComponent
         Item {
             id: wrapper
+            property int gapNowAndOldest: nextMessageIndex > -1 ? (Date.now() / 1000 - nextMsgTimestamp / 1000) : 0
+            property int gapNowAndJoined: Date.now() / 1000 - +chatsModel.channelView.activeChannel.joined / 1000
+
+            visible: {
+                return gapNowAndOldest < Constants.maxNbDaysToFetch * Constants.fetchRangeLast24Hours
+                        && gapNowAndJoined > Constants.maxNbDaysToFetch * Constants.fetchRangeLast24Hours
+                        && (chatsModel.activeChannel.chatType !== Constants.chatTypePrivateGroupChat || chatsModel.activeChannel.isMember)
```
Comment (emizzle) [inline ui/app/AppLayouts/Chat/ChatColumn/Message.qml:292]: Does this also work for communities (unsure if it is meant it not, so thought I'd mention it)?

File: ui/app/AppLayouts/Chat/ChatColumn/Message.qml [hunk 2]
```diff
@@ -283,14 +283,21 @@ Item {
         id: fetchMoreMessagesButtonComponent
         Item {
             id: wrapper
+            property int gapNowAndOldest: nextMessageIndex > -1 ? (Date.now() / 1000 - nextMsgTimestamp / 1000) : 0
+            property int gapNowAndJoined: Date.now() / 1000 - +chatsModel.channelView.activeChannel.joined / 1000
+
+            visible: {
+                return gapNowAndOldest < Constants.maxNbDaysToFetch * Constants.fetchRangeLast24Hours
+                        && gapNowAndJoined > Constants.maxNbDaysToFetch * Constants.fetchRangeLast24Hours
```
Comment (emizzle) [inline ui/app/AppLayouts/Chat/ChatColumn/Message.qml:291]: Could maybe pull this out in to a variable?

File: src/app/chat/views/chat_item.nim [hunk 3]
```diff
@@ -171,6 +171,13 @@ QtObject:
   proc muted*(self: ChatItemView): bool {.slot.} =
     return ?.self.chatItem.muted
 
+  proc joined*(self: ChatItemView): string {.slot.} = 
+    if self.chatItem.isNil: return ""
+    return $self.chatItem.joined
+
+  QtProperty[string] joined:
+    read = joined
+
```
Comment (emizzle) [inline src/app/chat/views/chat_item.nim:180]: What do you think about passing this to QML as `int` so that no parsing on QML side is needed?

**Review by emizzle — COMMENTED** [review]
Well done! Couple of small comments...

File: src/app/chat/views/chat_item.nim (hunk 3 above)
Comment (pinebit) [reply src/app/chat/views/chat_item.nim:180]: `joined` is int64
we may have this converted and exposed as seconds, but this would affect the precision

Comment (pinebit) [reply src/app/chat/views/chat_item.nim:180]: Also looks like we do this already with `nextMsgTimestamp` and `prevMsgTimestamp`

File: ui/app/AppLayouts/Chat/ChatColumn/Message.qml (hunk 2 above)
Comment (pinebit) [reply ui/app/AppLayouts/Chat/ChatColumn/Message.qml]: Fixed

**Review by iurimatias — APPROVED** [review]

**Pull request merged**